errai-internal-bom/pom.xml

```diff
@@ -302,6 +306,12 @@
         <artifactId>errai-api</artifactId>
         <version>${project.version}</version>
       </dependency>
+
+      <dependency>
+        <groupId>com.google.elemental2</groupId>
+        <artifactId>elemental2-dom</artifactId>
+        <version>${version.com.google.elemental2}</version>
+      </dependency>
```

Comment:
I think the place to add this dependency would be in `errai-bom` instead of `errai-internal-bom`. `errai-internal-bom` is supposed to contain the errai modules only. IIRC the elemental2 dependency is already listed on `errai-bom`.

Reply:
Thanks for the comment @tiagobento :-) However, I think that the `errai-bom` and the `errai-internal-bom` have different responsibilities.

Many projects using Errai import the `errai-internal-bom` ([drools-wb](https://github.com/kiegroup/drools-wb/blob/master/pom.xml#L73), [jbpm-wb](https://github.com/kiegroup/jbpm-wb/blob/master/pom.xml#L75), [dashbuilder](https://github.com/dashbuilder/dashbuilder/blob/master/pom.xml#L545), [kie-wb-distribution](https://github.com/kiegroup/kie-wb-distributions/blob/master/pom.xml#L66)).\

That allows the users of the Errai API to declare Errai modules and avoid illegal transitive dependencies ([see](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/pom.xml#L72)).

But now, that the Errai elements are deprecated, sounds fair enough to get the Elemental2 **form** Errai. I mean, I don't think that we should add the Elemental2 version in the `drools-wb` (or even, in `droolsjbpm-build-bootstrap`).

WDYT?

Reply:
Yeah.. I didn't know about this transitive dependencies check you guys have. I guess in this case it makes sense to have this dependency on the `errai-internal-bom`, but then I'd remove it from the `errai-bom`, given that then it would be duplicated inside Errai. 

I'd wait to hear what other people have to say about this matter, though.